catalogs/catalog-lakehouse-iceberg/src/main/java/com/datastrato/gravitino/catalog/lakehouse/iceberg/IcebergTable.java

```diff
@@ -60,6 +70,40 @@ public CreateTableRequest toCreateTableRequest() {
     return builder.build();
   }
 
+  /** Convert the distribution of Iceberg to the distribution of Gravitino. */
+  private void convertDistribution() {
+    if (null == distribution) {
+      properties.put(DISTRIBUTION_MODE, DistributionMode.NONE.modeName());
+    } else {
+      switch (distribution.strategy()) {
+        case HASH:
+          Preconditions.checkArgument(
+              ArrayUtils.isEmpty(distribution.expressions()),
+              "Iceberg's Distribution Mode.HASH does not support set expressions.");
+          Preconditions.checkArgument(
+              ArrayUtils.isNotEmpty(partitioning),
+              "Iceberg's Distribution Mode.HASH is distributed based on partition, but the partition is empty.");
+          properties.put(DISTRIBUTION_MODE, DistributionMode.HASH.modeName());
+          break;
+        case RANGE:
+          Preconditions.checkArgument(
+              ArrayUtils.isEmpty(distribution.expressions()),
+              "Iceberg's Distribution Mode.RANGE not support set expressions.");
+          Preconditions.checkArgument(
+              ArrayUtils.isNotEmpty(partitioning) && ArrayUtils.isNotEmpty(sortOrders),
```

Comment:
There is a gap between your description and implement,    we should check partitioning `and` sortOrders are empty? or just one is not empty?